channeldb/forwarding_log.go

```diff
@@ -78,25 +81,83 @@ type ForwardingEvent struct {
 	// AmtOut is the amount of the outgoing HTLC. Subtracting the incoming
 	// amount from this gives the total fees for this payment circuit.
 	AmtOut lnwire.MilliSatoshi
+
+	// IncomingHtlcID is the ID of the incoming HTLC in the payment circuit.
+	// If the HTLC ID is not set, the value will be nil.
+	IncomingHtlcID fn.Option[uint64]
+
+	// OutgoingHtlcID is the ID of the outgoing HTLC in the payment circuit.
+	// If the HTLC ID is not set, the value will be nil.
+	OutgoingHtlcID fn.Option[uint64]
 }
 
 // encodeForwardingEvent writes out the target forwarding event to the passed
 // io.Writer, using the expected DB format. Note that the timestamp isn't
 // serialized as this will be the key value within the bucket.
 func encodeForwardingEvent(w io.Writer, f *ForwardingEvent) error {
-	return WriteElements(
+	// First we write the required fields.
+	err := WriteElements(
 		w, f.IncomingChanID, f.OutgoingChanID, f.AmtIn, f.AmtOut,
 	)
+	if err != nil {
+		return err
+	}
+
+	// Then we write the optional HTLC IDs if they are set.
```

Comment:
To be more defensive - I would return an error here if `f.IncomingHtlcID.IsNone()` to make it clear that they are mandatory.